options/docker.go

```diff
@@ -27,14 +30,17 @@ func (opts *Docker) Validate() error {
 	catcher := grip.NewBasicCatcher()
 	catcher.NewWhen(opts.Port < 0, "port must be positive value")
 	catcher.NewWhen(opts.Image == "", "Docker image must be specified")
-	if opts.Platform == "" {
-		if PlatformSupportsDocker(runtime.GOOS) {
-			opts.Platform = runtime.GOOS
+	if opts.OS == "" {
```

Comment:
Nit: we can clean this up by setting a temporary variable with the OS (unless you really want the two separate error messages):
```
os := opts.OS
if os == "" {
        os = runtime.GOOS
}
if !OSSupportsDocker(os) {
        // handle error
}
```

Reply:
I think the separate error messages are preferable since you can distinguish between the invalid default and invalid explicitly-specified OS cases. This would also have to unconditionally assign the `os` back to `opts.OS` anyways. I don't feel that strongly, but it seems like the current way preserves more explicit error messages, even though the nested ifs are kind of ugly.